Src/ILGPU/Util/InlineList.cs

```diff
@@ -387,7 +388,7 @@ public void MoveTo(ref InlineList<T> list)
             int count,
             ref InlineList<T> list)
         {
-            list.Reserve(Count + count);
+            list.Reserve(Math.Max(list.Capacity, count));
             Array.Copy(items, startIndex, list.items, 0, count);
```

Comment:
ISSUE: At this point, `list.Count` may be incorrect. It should be fixed here before returning.

```suggestion
            Array.Copy(items, startIndex, list.items, 0, count);
            list.Count = Max.Math(list.Count, count);
```

Reply:
Definitely 👍 !